### DISCUSSION TITLE
Can not use debug with ftxui 2.0.0

### DISCUSSION THREAD

**Discussion by Leopold95** [ArthurSonzogni/FTXUI#329] [Q&A]
I cant use debug with ftxui 2 but can use relise
I cant use relise with ftxui 0.11 but can use debug

so what i should do for using 2.0.0 for debug and reliase ?

vs 2019 using c++ standart 20

**Comment by ArthurSonzogni** [comment]
Hello,

It was reported by the bot, FTXUI 2.0.0 was building in both release and debug mode on every OS:
https://github.com/ArthurSonzogni/FTXUI/actions/runs/1733518484

I don't use or own a Windows OS. So, I can't retrieve more informations

Could you please let me know what error you are encountering?

**Reply by Leopold95** [reply]
https://youtu.be/dacJYvQsINs
here is a private link on video which i did 
i think there will be enough 
watch it pls. i really want to know what i do wrong

**Reply by MattBystrin** [reply]
@Leopold95 as I can see you have linkage errors. Try to add folder that directly have .dll file.

Так же я из русско-говорящего сообщества. Можете смело писать мне на почту, я помогу вам разобраться ))

**Reply by Leopold95** [reply]
> 

@MattBystrin это хорошо. Что вы имеете введу ? Скажите подробнее пожалуйста.
(Еcли вы про проблемы с либами то это фиксится использованием pragma comment())

**Reply by MattBystrin** [reply]
@Leopold95 я ещё раз посмотрел видео. Возможно, некоторые конфигурации не работают, потому что библиотека компилировалась для 17 стандарта. Вы можете собрать проект с использованием 17 стандарта ? Если да, то проблема остается ?
UPD: В сети действительно есть информация, что для различных стандартов плюсов MSVC может генерить бинарники без обратной совместимости. В таком случае,  если Вы хотите использовать 20 стандарт, Вам придется собрать библиотеку самому

**Reply by ArthurSonzogni** [reply]
It is not recommended trying to specify how FTXUI is built and linked yourself.

Instead using CMake or vcpkg (which use cmake internally) is recommended.
Here is a starter project using CMake: https://github.com/ArthurSonzogni/ftxui-starter
visual studio should be able to open it directly by opening the CMakeLists.txt.

**Reply by Chasikanaft** [reply]
> @Leopold95 я ещё раз посмотрел видео. Возможно, некоторые конфигурации не работают, потому что библиотека компилировалась для 17 стандарта. Вы можете собрать проект с использованием 17 стандарта ? Если да, то проблема остается ? UPD: В сети действительно есть информация, что для различных стандартов плюсов MSVC может генерить бинарники без обратной совместимости. В таком случае, если Вы хотите использовать 20 стандарт, Вам придется собрать библиотеку самому

Сдается мне, это не в стандарте проблема, мне при билде debug/ x64 выдает LNK 2038 с несоответствием MD_DynamicRelease и MDd_DynamicDebug. Замена на другие MD результатов не дает, видимо из за постфикса Release, как выставить его в дебаг, не понимаю.

**Reply by Chasikanaft** [reply]
>Нашел решение, скачиваешь vcpkg устанавливаешь через него все что необходимо, profit.
